controllers/imapmon.go

```diff
@@ -0,0 +1,184 @@
+package controllers
+
+/* TODO:
+*		 - Have a counter per config for number of consecutive login errors and backoff (e.g if supplied creds are incorrect)
+*		 - Have a DB field "last_login_error" if last login failed
+*		 - DB counter for non-campaign emails that the admin should investigate
+*		 - Add field to User for numner of non-campaign emails reported
+ */
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/gophish/gophish/logger"
+
+	"github.com/glennzw/eazye"
```

Comment:
Yeah totally agree, additional 3rd party packages make us all nervous! 

So to be clear, [github.com/glennzw/eazye](github.com/glennzw/eazye) is my fork of [https://github.com/jprobinson/eazye](https://github.com/jprobinson/eazye) with added functionality (so it's not really my package). I agree with your idea to create a `github.com/gophish/gophish/imap` package to avoid nasties creeping in.

I'd like to move away from `eazye` at some point (as it uses outdated/unmaintained packages) and use `imap-go` but this was the quickest way to get up and running.

Reply:
I've created the `gophish\imap` package and moved functionality there. (I just copied in the single `eazye.go` file and added the `ValidateIMAP()` from the controller).

Also renamed `ImapMonitor` to just `Monitor`.

Next up: remove unused functionality in the new `imap.go` file and try use your `email.NewReader` to parse emails.

Commit: https://github.com/glennzw/gophish/commit/7ddba13507fab0fc2b5359afb929b9ec3793a201